KenticoCloud.Delivery/CodeFirst/ICodeFirstTypeProvider.cs

```diff
@@ -12,5 +12,11 @@ public interface ICodeFirstTypeProvider
         /// </summary>
         /// <param name="contentType">Content type identifier.</param>
         Type GetType(string contentType);
+
+        /// <summary>
+        /// Returns a codename corresponding to the given content type model.
+        /// </summary>
+        /// <param name="contentType">Content type model.</param>
+        string GetCodename(Type contentType);
```

Comment:
now you can rewrite or maybe even get rid of the `ContentTypeExtractor` class

Reply:
but make sure you keep the tests that check whether the `system.type` query param is added